fast_barcode_scanner/android/src/main/kotlin/com/jhoogstraat/fast_barcode_scanner/cache/ImageHelper.kt

```diff
@@ -0,0 +1,102 @@
+import android.content.Context
+import android.graphics.ImageFormat
+import android.graphics.Rect
+import android.graphics.YuvImage
+import android.media.Image
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+
+class ImageHelper {
+    private var savedCodes: MutableMap<String, String> = mutableMapOf()
+
+    private object Holder {
+        val INSTANCE = ImageHelper()
+    }
+
+    companion object {
+        @JvmStatic
+        fun getInstance(): ImageHelper {
+            return Holder.INSTANCE
+        }
+    }
+
+    // Store image to the path with barcode as filename
+    private fun storeImage(imageBytes: ByteArray?, key: String, context: Context) {
+        val externalFilesDirectory = context.cacheDir;
+        val imageFile: File
+
+        try {
+            externalFilesDirectory.mkdirs()
+            val barcodeDirectory = File(externalFilesDirectory, "barcode_images")
+            if (!barcodeDirectory.exists()) {
+                barcodeDirectory.mkdirs()
+            }
+
+            imageFile = File.createTempFile(key, ".jpeg", barcodeDirectory)
+            imageFile.deleteOnExit()
+        } catch (e: IOException) {
+            throw RuntimeException(e)
+        }
+
+        try {
+            FileOutputStream(imageFile).use { fos ->
+                fos.write(imageBytes)
+            }
+            savedCodes[key] = imageFile.absolutePath
+        } catch (e: IOException) {
+            e.printStackTrace()
+        }
+    }
+
+    private fun isImageSaved(code: String): Boolean {
+        return savedCodes.contains(code)
+    }
```

Comment:
we might not need this method since the `retrieveImagePath` method returns String?. If it's saved, then it will return, if not it will be null.

Reply:
This function prevents saving multiple images of the same code because ML Kit continuously detects the code when we point the camera at it. As a result, it outputs multiple frames with the same code. We only need to save one image per code